query/executor_test.go

```diff
@@ -595,6 +603,236 @@ func TestQueryExecutor_InvalidSource(t *testing.T) {
+	e.WithLogWriter(mockWatcher, context.Background())
+
+	e.StatementExecutor = &StatementExecutor{
+		ExecuteStatementFn: func(stmt influxql.Statement, ctx *query.ExecutionContext) error {
+			require.Equal(t, uint64(1), ctx.QueryID, "query ID")
+			return nil
+		},
+	}
+
+	discardOutput(e.ExecuteQuery(q, query.ExecutionOptions{}, nil))
+
+	require.True(t, mockWatcher.Contains(`SELECT count(value) FROM cpu`))
+}
+
+func TestQueryExecutor_IgnoreWriteEvent(t *testing.T) {
+	q, err := influxql.ParseQuery(`SELECT count(value) FROM cpu`)
+	require.NoError(t, err, "parse query")
+
+	e := NewQueryExecutor()
+	mockWatcher := newMockWatcher(t, "query.log")
+	e.WithLogWriter(mockWatcher, context.Background())
+
+	e.StatementExecutor = &StatementExecutor{
+		ExecuteStatementFn: func(stmt influxql.Statement, ctx *query.ExecutionContext) error {
+			require.Equal(t, uint64(1), ctx.QueryID, "query ID")
+			return nil
+		},
+	}
+
+	discardOutput(e.ExecuteQuery(q, query.ExecutionOptions{}, nil))
+
+	require.True(t, mockWatcher.Contains(`SELECT count(value) FROM cpu`))
+	require.Equal(t, uint64(0), mockWatcher.RemovedPaths.Load())
+	require.Equal(t, uint64(1), mockWatcher.AddedPaths.Load())
+	require.Equal(t, uint64(0), mockWatcher.ChangeEvents.Load())
+
+	mockWatcher.MockEvents <- fsnotify.Event{
+		Name: "Write",
+		Op:   fsnotify.Write,
+	}
+	// Should not register a write event as a change event
+	require.Equal(t, uint64(0), mockWatcher.ChangeEvents.Load())
+}
+
+func TestQueryExecutor_ChangeEvent(t *testing.T) {
+	q, err := influxql.ParseQuery(`SELECT count(value) FROM cpu`)
+	require.NoError(t, err, "parse query")
+
+	e := NewQueryExecutor()
+	mockWatcher := newMockWatcher(t, "query.log")
+	e.WithLogWriter(mockWatcher, context.Background())
+
+	e.StatementExecutor = &StatementExecutor{
+		ExecuteStatementFn: func(stmt influxql.Statement, ctx *query.ExecutionContext) error {
+			require.Equal(t, uint64(1), ctx.QueryID, "query ID")
+			return nil
+		},
+	}
+
+	discardOutput(e.ExecuteQuery(q, query.ExecutionOptions{}, nil))
+
+	require.True(t, mockWatcher.Contains(`SELECT count(value) FROM cpu`))
+	require.Equal(t, uint64(0), mockWatcher.RemovedPaths.Load())
+	require.Equal(t, uint64(1), mockWatcher.AddedPaths.Load())
+	require.Equal(t, uint64(0), mockWatcher.ChangeEvents.Load())
+
+	mockWatcher.MockEvents <- fsnotify.Event{
+		Name: "Rename",
+		Op:   fsnotify.Rename,
+	}
+
+	q, err = influxql.ParseQuery(`SELECT count(value) FROM mem`)
+
+	e.StatementExecutor = &StatementExecutor{
+		ExecuteStatementFn: func(stmt influxql.Statement, ctx *query.ExecutionContext) error {
+			require.Equal(t, uint64(1), ctx.QueryID, "query ID")
+			return nil
+		},
+	}
+
+	discardOutput(e.ExecuteQuery(q, query.ExecutionOptions{}, nil))
+
+	require.True(t, mockWatcher.Contains(`SELECT count(value) FROM mem`))
+
+	require.Equal(t, uint64(1), mockWatcher.RemovedPaths.Load())
+	require.Equal(t, uint64(1), mockWatcher.ChangeEvents.Load())
```

Comment:
I'm unsure why but this is return `uint64(2)` instead of the expected `uint64(1)` 

Above I have (for debugging I will remove it)
```go
func (m *mockWatcher) FileChangeCapture() error {
	fmt.Printf("change event! Current count: %d\n", m.ChangeEvents.Load())
	m.Remove(m.path)
	m.ChangeEvents.Add(1)
	fmt.Printf("After increment: %d\n", m.ChangeEvents.Load())
	return nil
}
```
The print out in the console while the test is running is

```
change event! Current count: 0
After increment: 1
```
So I am unsure why this is two...